allennlp/data/dataset_readers/text_classification_json.py

```diff
@@ -69,7 +69,13 @@ def _read(self, file_path):
                 text = items["text"]
                 label = items.get("label", None)
                 if label is not None:
-                    label = str(label)
+                    if self._skip_label_indexing:
+                        try:
+                            label = int(label)
+                        except ValueError:
+                            raise Exception('Labels must be integers if skip_label_indexing is True.')
```

Comment:
Should this be a `ConfigurationError` instead of a general Exception

Reply:
I made it a `ValueError` since I'm assuming the user intended to skip indexing, and that the input was incorrect.

Reply:
I don't have context on what the right error is here, just FYI on what the intent of `ConfigurationError` is: a `ConfigurationError` should be used when the problem is the result of something that was configured in a configuration file.